### ISSUE TITLE
Name change proposal for "Device" class to "Drone" or "Vehicle"

### ISSUE THREAD

**Issue by shakthi-prashanth-m** [dronecore/DroneCore#200]
This is regarding name change proposal of the class `Device`, `DeviceImpl`, etc.
We've named the library & API after **drone**, as in **DroneCore**, it makes sense to name its core service class to be `class Drone`, or at least `class Vehicle` (in QGroundControl) rather than `class Device`.

Cons of having `Device` name is that it is likely to be used for Ground Control Station, Mobile device also.
In other words, GCS is a device as well;
When we say `Vehicle`, there is no room of confusion whether it is Drone or Ground Control Station.

If DroneCore is also intended for general Robotics application, then probably `Device` might sound good.
Kindly share your thoughts, opinions.

Thanks.

**Comment by julianoes** [comment]
I chose device initially because DroneCore can also be used for a land robot or a (mavlink enabled) camera only.

**Comment by hamishwillee** [comment]
Device works for me given the camera-support aspect - and indeed there is no reason we couldn't do antenna trackers or any other mavlink controlled thingy.

**Comment by shakthi-prashanth-m** [comment]
Alright, if its used for standalone devices like camera, then Device is fine.
Thanks @julianoes @hamishwillee .
